annotation/annotation/annotations/main.py

```diff
@@ -110,8 +113,12 @@ def connect_s3(bucket_name: str) -> boto3.resource:
         # it responses with TypeError: NoneType object is not
         # callable, this try/except block should be changed after understanding
         # what is going on
-    except TypeError:
-        raise s3_resource.meta.client.exceptions.NoSuchBucket
+
+        # It throws S3 Client Error 404 now
+    except ClientError as e:
+        raise s3_resource.meta.client.exceptions.NoSuchBucket(
+            e.response, e.operation_name
```

Comment:
could you elaborate why pass these into NoSuchBucket exception?

Reply:
Without said arguments passed in from the ClientError, it throws a TypeError exception during the initialization of NoSuchBucket exception. 
[Issues covering this topic](https://github.com/boto/boto3/issues/3988)

> A HEAD request has the same options as a GET operation on an object. The response is identical to the GET response except that there is no response body. Because of this, if the HEAD request generates an error, it returns a generic code, such as 400 Bad Request, 403 Forbidden, 404 Not Found, 405 Method Not Allowed, 412 Precondition Failed, or 304 Not Modified. It's not possible to retrieve the exact exception of these error codes.

Reply:
But we raised NoSuchBucket exception before and it works just fine

Reply:
simply chain the exception from the caught one

Reply:
```
        except ClientError as e:
>           raise s3_resource.meta.client.exceptions.NoSuchBucket from e
E           TypeError: __init__() missing 2 required positional arguments: 'error_response' and 'operation_name'
```

This is the current problem. Even though I thought it worked, it seems not to.

Reply:
In which test you check for raise exception? Might be that you incorrectly mocked certain value

Reply:
I check that in test_connect_s3_no_bucket. The exception is raised from the connect_s3 method, not from the assert or pytest.raises.